### ISSUE TITLE
set ordering for parameters in reduction operation

### ISSUE THREAD

**Issue by nfarabullini** [GridTools/gt4py#844]
UnrollReduce always assumes that the first argument to the lifted reduce is a shifted call, e.g.

```rot_vec = neighbor_sum(vec_e(V2E) * geofac_rot, axis=V2EDim)```works but

```rot_vec = neighbor_sum(geofac_rot * vec_e(V2E) , axis=V2EDim)``` doesn't

Hence ```assert isinstance(node, ir.FunCall)```in the function does not pass

**Comment by DropD** [comment]
This assumption can be violated in two ways when lowering from FieldView currently (both execute fine in embedded mode):

1. As in the issue description the first field to occur in the `neighbor_sum` expression is a sparse field, so it doesn't need to be shifted. This lowers to 

```
call(
  fun=call(
    fun="lift", 
    args=[
      call(
        fun="reduce", 
        args=[(lambda init, x, y: init + (x * y)), 0]
      )
    ]
  ),
  args=[
    SymRef("geofac_rot"),  # <-- UnrollReduce asserts this to be a call with .fun.fun == "shift"...
    call(                  # ... but the shifted field is actually the second argument
      fun=call(
        fun="shift", 
        args=[OffsetLiteral("V2E")]
      ), 
      args=[SymRef("vec_e")]
    )
  ]
)
```

2. By using a temporary for the shifted field

```python
x = vec_e(V2E)
neighbor_sum(x, axis=V2EDim)
```

lowers to

```
call(
  fun=lambda "x": call(
    fun=call(
      fun="lift", 
      args=[call(
        fun="reduce",
        args=[(lambda init, xx: init + xx), 0]
      )]
    ),
    args=[SymRef("x")]  # <-- again, the assumption that this is a call(call("shift"), ...) is violated.
  ),
  args=[call(
    fun=call(fun="shift", args=[OffsetLiteral("V2E")]),  # ... the shift happens in the outer let expression and the reduction only contains a reference to it
    args=[SymRef("vec_e")]
  )]
)

**Comment by havogt** [comment]
Was resolved in #845.
